plugins/discourse-calendar/assets/javascripts/discourse/components/post-calendar.gjs

```diff
@@ -0,0 +1,377 @@
+import { action } from "@ember/object";
+import didInsert from "@ember/render-modifiers/modifiers/did-insert";
+import willDestroy from "@ember/render-modifiers/modifiers/will-destroy";
+import { service } from "@ember/service";
+import { htmlSafe } from "@ember/template";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import getURL from "discourse/lib/get-url";
+import { escapeExpression } from "discourse/lib/utilities";
+import Topic from "discourse/models/topic";
+import { colorToHex, contrastColor, stringToColor } from "../lib/colors";
+import FullCalendar from "./full-calendar";
+
+export default class PostCalendar extends Component {
+  @service currentUser;
+  @service siteSettings;
+  @service capabilities;
+  @service postCalendar;
+  @service router;
+  @service store;
+
+  @tracked post = this.args.post;
+
+  @action
+  registerPostCalendar() {
+    this.postCalendar.registerComponent(this);
+  }
+
+  @action
+  teardownPostCalendar() {
+    this.postCalendar.teardownComponent();
+  }
+
+  @action
+  onEventClick(info) {
+    if (info.event.extendedProps.postUrl) {
+      this.router.transitionTo(info.event.extendedProps.postUrl);
+    }
+  }
+
+  @action
+  async refresh() {
+    try {
+      const post = await this.store.find("post", this.post.id);
+      const topic_json = await Topic.find(post.topic_id, {});
+      const topic = Topic.create(topic_json);
+      post.set("topic", topic);
+      this.post = post;
+    } catch (error) {
+      popupAjaxError(error);
+    }
+  }
+
+  get isStatic() {
+    return this.args.options.calendarType === "static";
+  }
+
+  get isFullDay() {
+    return this.args.options.calendarFullDay === "true";
+  }
+
+  get timezone() {
+    let defaultTimezone = this.args.options.calendarDefaultTimezone;
+    const isValidDefaultTimezone = !!moment.tz.zone(
+      this.calendarDefaultTimezone
+    );
+
+    if (!isValidDefaultTimezone) {
+      defaultTimezone = null;
+    }
+
+    return defaultTimezone || this.currentUser?.timezone || moment.tz.guess();
+  }
+
+  get events() {
+    const events = [];
+
+    if (this.isStatic) {
+      events.push(...(this.args.staticEvents ?? []));
+    } else {
+      events.push(...(this.dynamicEvents ?? []));
+    }
+
+    return events;
+  }
+
+  get dynamicEvents() {
+    const events = [];
+    const groupedEvents = [];
+
+    (this.post.calendar_details || []).forEach((detail) => {
+      switch (detail.type) {
+        case "grouped":
+          if (this.isFullDay && detail.timezone) {
+            detail.from = moment
+              .tz(detail.from, detail.timezone)
+              .format("YYYY-MM-DD");
+          }
+          groupedEvents.push(detail);
+          break;
+        case "standalone":
+          if (this.isFullDay && detail.timezone) {
+            const eventDetail = { ...detail };
+            let from = moment.tz(detail.from, detail.timezone);
+            let to = moment.tz(detail.to, detail.timezone);
```

Comment:
```suggestion
            const from = moment.tz(detail.from, detail.timezone);
            const to = moment.tz(detail.to, detail.timezone);
```

Maybe? Unlesss `format` isn't allowed then because it mutates it